### ISSUE TITLE
Migrate Firefox to use WebDriver BiDi

### ISSUE THREAD

**Issue by jennifer-shehane** [cypress-io/cypress#29714]
### What would you like?

Firefox will be removing support for CDP at the end of 2024. [See announcement.](https://fxdx.dev/deprecating-cdp-support-in-firefox-embracing-the-future-with-webdriver-bidi/) Cypress will need to move all CDP over to WebDriver BiDi. 

We'll be able to defer the effects of the deprecation in Firefox 129 -[ See Issue](https://github.com/cypress-io/cypress/issues/29713)

Notably in the announcement: 
>The removal of CDP has no impact on Marionette, geckodriver and WebDriver BiDi. All these protocols will be maintained and further development happens to extend their feature sets.

### Other

This is a large issue so the work is being broken up into several smaller issues:
* https://github.com/cypress-io/cypress/issues/30217
* https://github.com/cypress-io/cypress/issues/30218
* https://github.com/cypress-io/cypress/issues/30219
* https://github.com/cypress-io/cypress/issues/30220
* https://github.com/cypress-io/cypress/issues/30221
* https://github.com/cypress-io/cypress/issues/30222

**Comment by whimboo** [comment]
Hello @AtofStryker. I wonder if this issue could be marked as done now given that CDP support is gone for Firefox.

**Comment by jennifer-shehane** [comment]
@whimboo Our migration is not fully complete. CDP support in Firefox will be removed in Firefox 141 officially, so we'll complete the migration before then.

**Comment by whimboo** [comment]
Oh, would you be ok with sharing what's left to do? I assume it's only work on Cypress internals and not related to the BiDi protocol then?

**Comment by AtofStryker** [comment]
Hey @whimboo. The only thing that is really left is migrating parts of the extension over to BiDi which I have a draft for [here](https://github.com/cypress-io/cypress/pull/31209). The issue for this is #30221. It's only related to Cypress internals to use the BiDi methods in place. We only need to keep the web extension around for a handful of methods which we talked about in the [State of Automation spreadsheet](https://docs.google.com/spreadsheets/d/1ED5y68I6RePTTIKLA6SsdP60qlXMUNFtNmNqhdWaeig/edit?gid=0#gid=0) and outlined in issue https://github.com/cypress-io/cypress/issues/30447. Though getting the extension cut over should allow us to close out our BiDi epic 🎉

**Comment by whimboo** [comment]
Ok, so based on the issue title it mentions `from CDP to WebDriver BiDi`. Maybe we could update it to say something like `Migrate Firefox to use WebDriver BiDi` or similar? Otherwise it gives (at least for me) the feeling that this issue only covers the CDP to BiDi migration part, which is not related to the remaining WebExtension work. But these are just my 2 cents.

**Comment by AtofStryker** [comment]
> Ok, so based on the issue title it mentions `from CDP to WebDriver BiDi`. Maybe we could update it to say something like `Migrate Firefox to use WebDriver BiDi` or similar? Otherwise it gives (at least for me) the feeling that this issue only covers the CDP to BiDi migration part, which is not related to the remaining WebExtension work. But these are just my 2 cents.

@whimboo good point. I just updated the title to more accurately reflect the work.
